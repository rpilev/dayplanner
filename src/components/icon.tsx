import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

import { SVGS } from '@app/images/svg'
import { IconInterfaces } from '@app/typescript/interfaces'

import SVG from 'react-inlinesvg'

const sizes: IconInterfaces.Sizes = {
  xs: 8,
  sm: 24,
  md: 48,
  lg: 84,
  xl: 128,
}

const IconWrapper = styled.div<IconInterfaces.IconWrapper>`
  ${({ size }) => css`
    width: ${sizes[size]}px;
    height: ${sizes[size]}px;
  `}
`

export const Icon = ({ name, size }: IconInterfaces.Icon): JSX.Element => (
  <IconWrapper size={size}>
    <SVG src={SVGS[name]} />
  </IconWrapper>
)

Icon.defaultProps = {
  size: 'md',
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
}
