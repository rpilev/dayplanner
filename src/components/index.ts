export * as Table from './table'
export * as Icon from './icon'
