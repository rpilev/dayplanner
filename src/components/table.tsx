import React, { ReactNode } from 'react'
import styled, { css } from 'styled-components'

import { TaskInterfaces } from '@app/typescript/interfaces'

const TableWrapper = styled.table`
  width: 100%;
  height: 100%;

  border-collapse: collapse;

  & > thead {
    font-size: 18px;
    color: #fff;
    text-align: left;
    border: 1px solid #000;

    & > tr {
      background: linear-gradient(0deg, rgba(22, 22, 22, 1) 0%, rgba(0, 0, 0, 1) 100%);
    }

    & > tr > th {
      padding: 15px 25px;
    }
  }

  & > tbody {
    color: #8c92a2;

    & > tr > td {
      border: 2px solid #1f2437;
      padding: 15px 35px;
    }

    & > tr:nth-child(odd) {
      background: linear-gradient(0deg, rgba(53, 60, 78, 1) 0%, rgba(52, 59, 77, 1) 100%);
    }

    & > tr:nth-child(even) {
      background: linear-gradient(0deg, rgba(45, 52, 70, 1) 0%, rgba(44, 51, 69, 1) 100%);
    }
  }
`

export const Th = styled.th<TaskInterfaces.Th>`
  ${({ tiny }) =>
    tiny &&
    css`
      width: 1%;
    `}
`

export const Td = styled.td<TaskInterfaces.Td>`
  ${({ center }) =>
    center &&
    css`
      text-align: center;
    `}

  ${({ fixedWidth }) =>
    fixedWidth &&
    css`
      width: ${fixedWidth}px;
      max-width: ${fixedWidth}px;

      overflow: hidden;
      text-overflow: ellipsis;
    `}

  ${({ oneLine }) =>
    oneLine &&
    css`
      max-width: 0;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `}

  ${({ multiLine }) =>
    multiLine &&
    css`
      word-break: break-word;
    `}
`

export const Table = ({ children }: { children: ReactNode }): JSX.Element => (
  <TableWrapper>{children}</TableWrapper>
)
