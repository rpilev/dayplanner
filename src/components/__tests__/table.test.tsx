import React from 'react'
import { render } from '@testing-library/react'

import { Table } from '@app/components'

interface Task {
  id: number
  column1: string
  column2: string
}

let tasks: Task[]

beforeEach(() => {
  tasks = [
    {
      id: 1,
      column1: 'column1 value 1',
      column2: 'column2 value 1',
    },
    {
      id: 2,
      column1: 'column1 value 2',
      column2: 'column2 value 2',
    },
    {
      id: 3,
      column1: 'column1 value 3',
      column2: 'column2 value 3',
    },
  ]
})

describe('Table', () => {
  test('should render correctly', () => {
    const IconRender = render(
      <Table.Table>
        <thead>
          <tr>
            <Table.Th>Title 1</Table.Th>
            <Table.Th>Title 2</Table.Th>
          </tr>
        </thead>
        <tbody>
          {tasks.map(({ id, column1, column2 }) => (
            <tr key={id}>
              <Table.Td>{column1}</Table.Td>
              <Table.Td>{column2}</Table.Td>
            </tr>
          ))}
        </tbody>
      </Table.Table>,
    )

    expect(IconRender).toBeDefined()
    expect(IconRender).toMatchSnapshot()
  })
})
