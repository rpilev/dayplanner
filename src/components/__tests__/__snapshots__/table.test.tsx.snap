// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-bdnylx geRHdX"
      >
        <thead>
          <tr>
            <th
              class="sc-gtssRu lbqNjK"
            >
              Title 1
            </th>
            <th
              class="sc-gtssRu lbqNjK"
            >
              Title 2
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column1 value 1
            </td>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column2 value 1
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column1 value 2
            </td>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column2 value 2
            </td>
          </tr>
          <tr>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column1 value 3
            </td>
            <td
              class="sc-dlnjPT kqlUpZ"
            >
              column2 value 3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-bdnylx geRHdX"
    >
      <thead>
        <tr>
          <th
            class="sc-gtssRu lbqNjK"
          >
            Title 1
          </th>
          <th
            class="sc-gtssRu lbqNjK"
          >
            Title 2
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column1 value 1
          </td>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column2 value 1
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column1 value 2
          </td>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column2 value 2
          </td>
        </tr>
        <tr>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column1 value 3
          </td>
          <td
            class="sc-dlnjPT kqlUpZ"
          >
            column2 value 3
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
